apps/dpnk/fakturoid_invoice_gen.py

```diff
@@ -110,151 +240,264 @@ def create_invoice_lines(invoice):
     """
     error_message = (
-        "Get Fakturoid API instance wasn't succesfull with Fakturoid"
-        " account '{fakturoid_user_account_slug}' due error: '{error}'"
+        "Obtaining access token wasn't successful with Fakturoid"
+        " client_id '{client_id}' due error: '{error}'"
     )
+
+    # Create the authorization header using HTTP Basic auth
+    auth_string = f"{client_id}:{client_secret}"
+    auth_bytes = auth_string.encode('ascii')
+    auth_b64 = base64.b64encode(auth_bytes).decode('ascii')
+
+    # Common headers for all requests
+    headers = {
+        'Accept': 'application/json',
+        'Authorization': f'Basic {auth_b64}'
+    }
+
+    data = {
+        'grant_type': "client_credentials",
+    }
+
+    # Try with JSON content type first
+    json_headers = headers.copy()
+    json_headers['Content-Type'] = 'application/json'
+
+    response = requests.post(
+        'https://app.fakturoid.cz/api/v3/oauth/token',
+        json=data,
+        headers=json_headers
+    )
+
     try:
-        fa = Fakturoid(
-            slug=settings.FAKTUROID[account]["user_account"],
-            email=settings.FAKTUROID[account]["user_email"],
-            api_key=settings.FAKTUROID[account]["api_key"],
-            user_agent=settings.FAKTUROID[account]["user_agent_header"],
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logger.error(
+            error_message.format(
+                client_id=client_id,
+                error=str(e)
+            )
         )
-    except (
-        requests.RequestException,
-        requests.ConnectionError,
-        requests.HTTPError,
-        requests.URLRequired,
-        requests.TooManyRedirects,
-        requests.Timeout,
-    ) as error:
+        return None
+
+    token_data = response.json()
+
+    return token_data['access_token']
+
+
+def create_api_session(account):
+    """Create authenticated session for Fakturoid API V3 using OAuth
+
+    :param str account: Fakturoid account type "production" or "test"
+
+    :return tuple: (session, base_url) or (None, None)
+    """
+    error_message = (
+        "Creating Fakturoid API session wasn't successful with Fakturoid"
+        " account '{fakturoid_account}' due error: '{error}'"
+    )
+
+    try:
+        # Get access token
+        access_token = obtain_access_token(
+            client_id=settings.FAKTUROID[account]["client_id"],
+            client_secret=settings.FAKTUROID[account]["client_secret"],
+        )
+        
+        if not access_token:
+            return None, None
+
+        # Create session with OAuth Bearer token
+        session = requests.Session()
+        session.headers.update({
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {access_token}"
+        })
+
+        slug = settings.FAKTUROID[account]["user_account"]
+        base_url = f"https://app.fakturoid.cz/api/v3/accounts/{slug}"
```

Comment:
Could you use global Fakturoid settings key for REST API base URL `FAKTUROID["base_rest_api_url"]` instead of hardcoded URL string, please? Old Faktruroid V2 API use `FAKTUROID_BASE_REST_API_URL` ENV with value https://app.fakturoid.cz/api/v2.

Project Fakturoid settings:

https://github.com/auto-mat/do-prace-na-kole/blob/6a0c1968a235c11ea88c4f30a8906188e66c1227/project/settings/base.py#L907-L909

K8 DPNK app config map:

https://github.com/auto-mat/k8s/blob/bfa74bf3ee2532793177a04d918c9f2f0207e79c/manifests/config-maps/dpnk.yaml#L29